Source/JavaScriptCore/runtime/JSString.cpp

```diff
@@ -355,4 +355,78 @@ JSString* jsStringWithCacheSlowCase(VM& vm, StringImpl& stringImpl)
     return string;
 }
 
+JSString* JSString::replaceOneCharImpl(JSGlobalObject* globalObject, UChar search, JSString* replacement, bool& found, unsigned recursionLimit, uint8_t* stackLimit)
+{
+    if (UNLIKELY(bitwise_cast<uint8_t*>(currentStackPointer()) < stackLimit || !recursionLimit))
+        return nullptr;
+    recursionLimit--;
```

Comment:
We probably do not need to have recursionLimit since we are checking real stack.